book.rb

```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
```

Comment:
* Since The attribute @rentals is what holds the relationship and to manage it we will also create a method to add rentals to it.
* Note ✍️ `The Book itself should be sent as a parameter to create the new rental achieving the 'has-many' association` [Kindly read this article on many-to-many relationship](
`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`)

Reply:
The Book is already being sent as a parameter to constructor of the rental
```js
class Rental
  attr_accessor :date
  attr_reader :book, :person

  def initialize(date, book, person)
    @date = date

    @book = book
    book.rentals << self

    @person = person
    person.rentals << self
  end
...
```

Reply:
* @usmansbk this only the association belongs-to-a-book and person

Reply:
@Davidosky007 That's exactly how it is done in the lesson, and It works, and I'm still unable to find the instructions you're referring to. Is this an optional change or required?

Reply:
* This only addresses the belongs-to association and not the has many association [Read this article ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

Reply:
@Davidosky007 Rental is not analogous Animal. Rental is analogous to Visit

Reply:
@Davidosky007 while Student is analogous to Animal in the example

Reply:
@Davidosky007 and Vet is to Classroom while Person is the Owner

Reply:
* All that's needed here is a method for adding rentals like you did for the Classroom class 
![image](https://user-images.githubusercontent.com/52497246/135835347-db8c27dd-218c-4005-88e4-0e57ce084e22.png)

Reply:
@Davidosky007 an error occured. `person` in Rental class is `read only`. So `rental.person = self` throws an error.